read_utils.py

```diff
@@ -1067,12 +1068,42 @@ def align_and_fix(
         log.warn("are you sure you meant to do nothing?")
         return
 
+    assert aligner in ["novoalign", "bwa"]
+
+    bwa_map_threshold = 30
+
+    tools.picard.CreateSequenceDictionaryTool().execute(refFasta, overwrite=True)
+    tools.samtools.SamtoolsTool().faidx(refFasta, overwrite=True)    
+
+    if aligner_options is None:
+        if aligner=="novoalign":
+            aligner_options = '-r Random'
+        elif aligner=='bwa':
+            aligner_options = '-T ' + str(bwa_map_threshold)
+
     bam_aligned = mkstempfname('.aligned.bam')
-    tools.novoalign.NovoalignTool().execute(
-        inBam, refFasta, bam_aligned,
-        options=novoalign_options.split(),
-        JVMmemory=JVMmemory
-    )
+    if aligner=="novoalign":
+        
+        tools.novoalign.NovoalignTool().index_fasta(refFasta)
+
+        tools.novoalign.NovoalignTool().execute(
+            inBam, refFasta, bam_aligned,
+            options=aligner_options.split(),
+            JVMmemory=JVMmemory
+        )
+    elif aligner=='bwa':
+        bwa = tools.bwa.Bwa()
+        bwa.index(refFasta)
+
+        aln_bam_prefilter = util.file.mkstempfname('.prefiltered.bam')
+        bwa.mem(inBam, refFasta, aln_bam_prefilter, opts=aligner_options.split()+['-R', '@RG\tID:bwa\tSM:sample'])
```

Comment:
Perfect; thanks for the comment. Forcing a read group stood out as incorrect, and I had hoped you'd weigh in.